[engenhui1999] iP

engenhui1999 opened the pull request:
# TaskGenie
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

TaskGenie frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use
- All you need to do is,


download it from [here](https://github.com/engenhui1999/ip/releases/tag/v1.0).

double-click it.
add your tasks.
let it manage your tasks for you 😉
And it is **FREE**! 😱 

Features:

- [X] Managing tasks
- [X] Managing deadlines
- [X] Managing events
- [ ] Reminders (coming soon)
---------------
If you Java programmer, you can use it to practice Java too. Here's the main method:

```
public class Main {
    public static void main(String[] args) {
        Launcher.main(args);
    }
}
```

Commit:
Add Gradle support

engenhui1999 committed:
Modified Duke.java for Level-1

engenhui1999 committed:
Added Level-2

engenhui1999 committed:
Added Level-3

engenhui1999 committed:
Added Level-4

engenhui1999 committed:
Adding Task files as I have accidentally omitted them earlier.

engenhui1999 committed:
Added A-TextUiTesting.

engenhui1999 committed:
Added Level5-1.

engenhui1999 committed:
Added Level-6

engenhui1999 committed:
Added A-Enums.

engenhui1999 committed:
Fixed a mistake in the printing of error messages.

engenhui1999 committed:
Created packages for neater organisation.

engenhui1999 committed:
Added Level-7.

engenhui1999 committed:
Added Level-8.

engenhui1999 committed:
Merge branch 'branch-Level-8'

engenhui1999 committed:
Modified the code for Event and Deadline to accomodate the merge.

engenhui1999 committed:
Updated with better references to coding standards.

engenhui1999 committed:
Added A-MoreOOP.

engenhui1999 committed:
Merge branch 'add-gradle-support'

engenhui1999 committed:
Added A-Gradle.

engenhui1999 committed:
Added A-JUnit.

engenhui1999 committed:
Added A-JavaDoc.

engenhui1999 committed:
Added Level-9.

engenhui1999 committed:
Merge branch 'branch-Level-9'

engenhui1999 committed:
Modified the code to comply with the coding standards.

engenhui1999 committed:
Modified the code for a better abstraction barrier.

engenhui1999 committed:
Modified the package name to comply with coding standards.

kennethk-1201 commented on src/main/java/Parser/Parser.java:
```diff
@@ -6,7 +6,7 @@
 
 public class Parser {
     protected static enum CommandList {
```
I think your use of a command list is really neat!

kennethk-1201 commented on src/main/java/Storage/TaskList.java:
```diff
@@ -43,9 +43,26 @@ public boolean isEmpty() {
+        String str = "";
+        for (int i = 1; i <= this.db.size(); i++) {
+            if (this.db.get(i - 1).getDescription().contains(description)) {
+                str += (i + "." + this.db.get(i - 1) + "\n");
```
Could this line possibly be abstracted out as a function?

engenhui1999 replied:
Thank you for the suggestion! I will take note of this in my future commits!

engenhui1999 committed:
Fixed a bug in TodoCommand.

engenhui1999 committed:
Modified the code to fix bugs regarding invalid inputs.

engenhui1999 committed:
Added Level-10.

kynhan commented on src/main/java/Storage/TaskList.java:
```diff
@@ -0,0 +1,94 @@
+     */
+    public boolean isEmpty() {
+        return this.db.isEmpty();
+    }
```
I like the use of the "is" prefix for this boolean method.

kynhan commented on src/main/java/Storage/TaskList.java:
```diff
@@ -0,0 +1,94 @@
+     * @return A boolean value based on whether the database is empty.
+     */
+    public boolean isEmpty() {
+        return this.db.isEmpty();
```
Maybe instead of using this.db, you can just use db (like you did on line 29)? I think it would make the code neater and more concise.

kynhan reviewed (commented):
Overall, code is very neat and clean. 
Some minor things to consider only.

engenhui1999 committed:
Modified the JavaFX GUI to run Duke, and fixed some bugs with storing of the database.

engenhui1999 committed:
Modified and added more OOP to the code and fixed checkstyle errors.

engenhui1999 committed:
Command classes: Add assert methods to document important assumption that should hold true.

Command classes: Have always assumed that correct parameters are given when calling the classes.

Having assertions will allow us to document important assumptions that should hold true which
can be used for bug tracking and validity checking.

engenhui1999 committed:
Deadline and Event classes: Extract out the a common method between the deadline and event class.

Deadline and Event classes: Changing of string to LocalDateTime have some common behaviours in both classes.

The common behaviours across the two classes casue code duplication. Extracting it will allow
me to use polymorphism more when converting string into LocalDateTime.

engenhui1999 committed:
Merge pull request #1 from engenhui1999/branch-A-Assertions

Approved

engenhui1999 committed:
Merge pull request #2 from engenhui1999/branch-A-CodeQuality

Approved

engenhui1999 committed:
Add A-CI to ensure that production workflow stays the same with new commits.

engenhui1999 committed:
Merge branch 'master' of https://github.com/engenhui1999/ip

engenhui1999 committed:
UpdateCommand Class: Add UpdateCommand to allow for editing of any details of a task.

The user may have incorrectly typed the details of a task and wish to edit the details.
This class allows for editing of the details without deleting the task.

engenhui1999 committed:
Modify code to comply with coding standards

engenhui1999 committed:
Modify gradle.yml to better fit my project

engenhui1999 committed:
Modify gradle.yml to better fit my project

engenhui1999 committed:
Modify the runtest.sh due to compilation problems

engenhui1999 committed:
Modify runtest.sh due to compilation problems

engenhui1999 committed:
Modify runtest.sh for compilation problems

engenhui1999 committed:
Modify gradle.yml to only check for Windows

engenhui1999 committed:
Modify gradle.yml to better fit my project

engenhui1999 committed:
Merge branch 'master' into branch-C-Update

engenhui1999 committed:
Revert gradle.yml to accomodate for Linux and MacOS uses

engenhui1999 committed:
Removed all I/O redirection as it is creating errors

engenhui1999 committed:
Merge pull request #4 from engenhui1999/branch-C-Update

Approved

engenhui1999 committed:
Modify documentation due to grammatical errors

engenhui1999 committed:
Modify build.gradle for building of Jar file

engenhui1999 committed:
Add unique personality

Refactor all names that contain Duke to TaskGenie and added new pictures to suit TaskGenie

engenhui1999 committed:
Fix error as the package name was incorrect

engenhui1999 committed:
Fix code to follow closer to the code standards

engenhui1999 committed:
Release User Guide for using the TaskGenie

engenhui1999 committed:
Fix formatting issues for the user guide

engenhui1999 committed:
Change pictures of the TaskGenie

engenhui1999 committed:
Fix documentation errors

engenhui1999 committed:
Change directory of folder to fix naming conventions

engenhui1999 committed:
Fix naming conventions

engenhui1999 committed:
Change directory of folder to fix naming conventions

engenhui1999 committed:
Fix naming conventions

engenhui1999 committed:
Fix documentation errors

engenhui1999 committed:
Fix truncating issues in JavaFX

engenhui1999 committed:
Fix formatting errors
